examples/esp-chat-react/js/model/eventProcessors/ChatAppEventProcessor.js

```diff
@@ -0,0 +1,60 @@
+"use strict";
+
+var esp = require('esp-js');
+var MessageSubscription = require('./MessageSubscription');
+
+var ChatAppEventProcessor = function (router, modelId) {
+    esp.model.DisposableBase.call(this);
+    this.router = router;
+    this.modelId = modelId;
+    this.messageSubscription = null;
+};
+
+ChatAppEventProcessor.prototype = Object.create(esp.model.DisposableBase.prototype);
+
+ChatAppEventProcessor.prototype.start = function () {
+    this.messageSubscription = new MessageSubscription(this.router, this.modelId);
+    this.addDisposable(this.messageSubscription);
+    this.observeInitEvent();
+    this.observeThreadSelected();
+    this.observeMessagesReceived();
+};
+
+ChatAppEventProcessor.prototype.observeInitEvent = function () {
+    this.addDisposable(this.router
+        .getEventObservable(this.modelId, "initEvent")
+        .observe(function () {
+            this.messageSubscription.start();
+        }.bind(this))
```

Comment:
I believe it's fine here. 

Just to avoid ambiguity `.bind(this)` is required in order to bind `this` within the function passed to `observe()`, else the `this` in that function would be bound to something else. 

Being on the proto def shouldn't matter as that `bind()` only happens when `.observeInitEvent()` is invoked against a speific instance of `ChatAppEventProcessor` and in that case `this` will be that instance itself (assuming it's called in the normal manor: `myInstance.observeInitEvent()`. So each call to this `observeInitEvent()` will be acting on it's own state.